Sources/Datadog/Tracing/Propagation/OpenTelemetry/OpenTelemetryHTTPHeadersWriter.swift

```diff
@@ -0,0 +1,135 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+/// The `OpenTelemetryHTTPHeadersWriter` should be used to inject trace propagation headers to
+/// the network requests send to the backend instrumented with Datadog APM.
+/// The injected headers conform to [Open Telemetry](https://github.com/openzipkin/b3-propagation) standard.
+///
+/// Usage:
+///
+///     var request = URLRequest(...)
+///
+///     let writer = OpenTelemetryHTTPHeadersWriter(openTelemetryHeaderType: .single)
+///     let span = Global.sharedTracer.startSpan("network request")
+///     writer.inject(spanContext: span.context)
+///
+///     writer.tracePropagationHTTPHeaders.forEach { (field, value) in
+///         request.setValue(value, forHTTPHeaderField: field)
+///     }
+///
+///     // call span.finish() when the request completes
+///
+///
+public class OpenTelemetryHTTPHeadersWriter: OTHTTPHeadersWriter {
+    /// Open Telemetry header type.
+    ///
+    /// There are two encodings of B3:
+    /// [Single Header](https://github.com/openzipkin/b3-propagation#single-header)
+    /// and [Multiple Header](https://github.com/openzipkin/b3-propagation#multiple-headers).
+    ///
+    /// Multiple header encoding uses an `X-B3-` prefixed header per item in the trace context.
+    /// Single header delimits the context into into a single entry named b3.
+    /// The single-header variant takes precedence over the multiple header one when extracting fields.
+    public enum OpenTelemetryHeaderType {
+        case multiple, single
+    }
```

Comment:
[@opentelemetry/propagator-b3](https://www.npmjs.com/package/@opentelemetry/propagator-b3) uses `injectEncoding` instead of type. Also, it is already inner to `OpenTelemetryHTTPHeadersWriter`, so no need to add a prefix :)

Reply:
You suggest we change the name or move away from enum to some consts as well? 🤔

Reply:
No no, just about naming. We could reuse `InjectEncoding`. But it's just a suggestion ;)